rollup/rollup_sync_service/rollup_sync_service.go

```diff
@@ -509,69 +503,56 @@ func validateBatch(batchIndex uint64, event *L1FinalizeBatchEvent, parentFinaliz
 		Chunks:                     chunks,
 	}
 
-	var codecVersion *uint8
+	var codecVersion encoding.CodecVersion
 	if committedBatchMeta != nil {
-		codecVersion = &committedBatchMeta.Version
+		codecVersion = encoding.CodecVersion(committedBatchMeta.Version)
+	} else {
+		codecVersion = encoding.GetCodecVersion(chainCfg, startBlock.Header.Number, startBlock.Header.Time)
 	}
-	determinedCodecVersion := determineCodecVersion(startBlock.Header.Number, startBlock.Header.Time, chainCfg, codecVersion)
 
 	var localBatchHash common.Hash
-	if determinedCodecVersion == encoding.CodecV0 {
-		daBatch, err := codecv0.NewDABatch(batch)
-		if err != nil {
-			return 0, nil, fmt.Errorf("failed to create codecv0 DA batch, batch index: %v, err: %w", batchIndex, err)
-		}
-		localBatchHash = daBatch.Hash()
-	} else if determinedCodecVersion == encoding.CodecV1 {
-		daBatch, err := codecv1.NewDABatch(batch)
-		if err != nil {
-			return 0, nil, fmt.Errorf("failed to create codecv1 DA batch, batch index: %v, err: %w", batchIndex, err)
-		}
-		localBatchHash = daBatch.Hash()
-	} else if determinedCodecVersion == encoding.CodecV2 {
-		daBatch, err := codecv2.NewDABatch(batch)
-		if err != nil {
-			return 0, nil, fmt.Errorf("failed to create codecv2 DA batch, batch index: %v, err: %w", batchIndex, err)
-		}
-		localBatchHash = daBatch.Hash()
-	} else if determinedCodecVersion == encoding.CodecV3 {
-		daBatch, err := codecv3.NewDABatch(batch)
+
+	codec, err := encoding.GetCodec(codecVersion)
+	if err != nil {
+		return 0, nil, fmt.Errorf("unsupported codec version: %v, batch index: %v, err: %w", codecVersion, batchIndex, err)
+	}
+
+	if codecVersion != encoding.CodecV4 {
+		daBatch, err := codec.NewDABatch(batch)
 		if err != nil {
-			return 0, nil, fmt.Errorf("failed to create codecv3 DA batch, batch index: %v, err: %w", batchIndex, err)
+			return 0, nil, fmt.Errorf("failed to create DA batch, batch index: %v, codec version: %v, err: %w", batchIndex, codecVersion, err)
 		}
 		localBatchHash = daBatch.Hash()
-	} else if determinedCodecVersion == encoding.CodecV4 {
-		// Check if committedBatchMeta exists, for backward compatibility with older client versions
+	} else {
```

Comment:
I think we can put below codes to da-codec repo and add function like `NewDABatchWithBlobVersionedHashes` to the codec interface.

Reply:
makes sense.

Reply:
changed to use `NewDABatchWithBlobVersionedHashes`.